### PR TITLE
Upgrade prometheus to 2.22.2

### PR DESCRIPTION
**Pull request by branden**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:

This upgrades Prometheus to 2.22.2.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
[prometheus] Upgrades Prometheus to 2.22.2.
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: addons/prometheus/prometheus.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ metadata:
   annotations:
     catalog.kubeaddons.mesosphere.io/addon-revision: "0.38.1-22"
```
Comment (gracedo) [inline addons/prometheus/prometheus.yaml:13]:
```suggestion
    catalog.kubeaddons.mesosphere.io/addon-revision: "0.38.1-23"
```

**Review by gracedo — APPROVED** [review]

**Comment by branden** [comment]
Looks like the generate-pipeline check failed due to a flake. I re-ran it and it succeeded: https://konvoy-staging.production.d2iq.cloud/dispatch/dashboard/ci/pipelines/namespace/dispatch/kubernetes-base-addons/runs/gen-mesosphere-kubernetes-base-addons-pr-681-f3eac23-7gh8nxwrsv/logs

**Review by gracedo — APPROVED** [review]

File: addons/prometheus/prometheus.yaml [hunk 2]
```diff
@@ -206,7 +206,7 @@ spec:
                   insecureSkipVerify: true
         prometheusSpec:
           image:
-            tag: v2.19.2
+            tag: v2.22.2
```
Comment (samvantran) [inline addons/prometheus/prometheus.yaml:209]: Anything noteworthy in these releases in terms of notes for our users?

Comment (branden) [reply addons/prometheus/prometheus.yaml:209]: Lots of new fixes and features, but I didn't see anything that looked like it should be surfaced in KBA release notes.

**Review by samvantran — APPROVED** [review]

**Comment by sebbrandt87** [comment]
are we sure that it works in all cases? I have seen problems updating the prometheus targets, rules etc

**Comment by branden** [comment]
Thanks @sebbrandt87, good catch. I have a PR to revert the upgrade: https://github.com/mesosphere/kubernetes-base-addons/pull/737

**Pull request merged**